unstructured/metrics/evaluate.py

```diff
@@ -327,32 +360,79 @@ def measure_table_structure_accuracy(
                 ["element_col_level_content_acc", None, None, None, 0],
                 ["element_row_level_content_acc", None, None, None, 0],
             ]
-        ).reset_index()
+        ).transpose()
     else:
-        element_metrics_results = {}
-        for metric in [
-            "total_tables",
-            "table_level_acc",
-            "element_col_level_index_acc",
-            "element_row_level_index_acc",
-            "element_col_level_content_acc",
-            "element_row_level_content_acc",
-        ]:
-            metric_df = has_tables_df[has_tables_df[metric].notnull()]
-            agg_metric = metric_df[metric].agg([_mean, _stdev, _pstdev, _count]).transpose()
-            if agg_metric.empty:
-                element_metrics_results[metric] = pd.Series(
-                    data=[None, None, None, 0], index=["_mean", "_stdev", "_pstdev", "_count"]
-                )
-            else:
-                element_metrics_results[metric] = agg_metric
-        agg_df = pd.DataFrame(element_metrics_results).transpose().reset_index()
+        # filter out documents with no tables
+        having_table_df = df[df["total_tables"] > 0]
```

Comment:
If we're touching this - it would be good to safely remove this condition. It checks if ground truth file has some tables. But it's possible that it does not contain tables, but we'll find some (false positive). At the same time we should cover various scenarios when data contains NaNs or is empty at all.